code/scripting/api/libs/base.cpp

```diff
@@ -61,26 +62,27 @@ ADE_FUNC(rand32,
 
 	int result;
 	if (numargs == 2)
-		result = rand32(a, b);
+		result = ::util::Random::next(a, b);
 	else if (numargs == 1)
-		result = rand32() % a;
+		result = ::util::Random::next(a);
 	else
-		result = rand32();
+		result = ::util::Random::next();
 
 	return ade_set_error(L, "i", result);
 }
 
 ADE_FUNC(rand32f,
 	l_Base,
 	"[number max]",
-	"Calls FSO's rand32() function and transforms the result to a float.  If called with no arguments, returns a random float from [0.0, 1.0).  If called with one argument, returns a float from [0.0, max).",
+	"Calls FSO's Random::next() function and transforms the result to a float.  If called with no arguments, returns a random float from [0.0, 1.0).  If called with one argument, returns a float from [0.0, max).",
 	"number",
 	"A random float")
 {
 	float _max;
 	int numargs = ade_get_args(L, "|f", &_max);
 
-	float result = static_cast<float>(rand32()) / static_cast<float>(0x7fffffff);
+	// DISCUSSME: is division at the level of 2**31 ok for floats?
+	float result = static_cast<float>(::util::Random::next()) / static_cast<float>(::util::Random::MAX_VALUE);
```

Comment:
yup and my experiments with `float` have shown that `float` is precise enough that it can handle `1.0/(2**31-1)` (to put it in Python). I'll use `RAND_MAX_1f` instead of division, since that's what's done elsewhere.